lib/simple_form/inputs/base.rb

```diff
@@ -72,11 +72,14 @@ def initialize(builder, attribute_name, column, input_type, options = {})
         @html_classes = SimpleForm.additional_classes_for(:input) { additional_classes }
 
         @input_html_classes = @html_classes.dup
-        if SimpleForm.input_class && !input_html_classes.empty?
-          input_html_classes << SimpleForm.input_class
+
+        input_html_classes_memoized = input_html_classes
+
+        if SimpleForm.input_class && !input_html_classes_memoized.empty?
```

Comment:
Since we are here, we could use `input_html_classes.any?` rather than `!input_html_classes.empty?`.